### PR TITLE
Add checkpoint conversion instructions to GRPO docs

### PR DESCRIPTION
**Pull request by khatwanimohit**
# Description

- add environment variable section 
- add checkpoint conversion instructions 

If the change fixes a bug or a Github issue, please include a link, e.g.,:
FIXES: b/123456
FIXES: #123456

*Notice 1:* Once all tests pass, the "pull ready" label will automatically be assigned.
This label is used for administrative purposes. Please do not add it manually.

*Notice 2:* For external contributions, our settings currently require an approval from a MaxText maintainer to trigger CI tests.

# Tests

n/a

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: docs/tutorials/grpo_with_pathways.md [hunk 1]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
+export MODEL=<model name> # e.g., 'llama3.1-70b'
+export TOKENIZER=<tokenizer path> # e.g., 'meta-llama/Llama-3.1-70B-Instruct'
+export HF_TOKEN=<Hugging Face access token>
+
+# -- MaxText configuration --
+export BASE_OUTPUT_DIRECTORY=<output directory to store run logs> # e.g., gs://my-bucket/my-output-directory
+export RUN_NAME=<name for this run> # e.g., $(date +%Y-%m-%d-%H-%M-%S)
+export MAXTEXT_CKPT_PATH=${BASE_OUTPUT_DIRECTORY}/${RUN_NAME}/0/items
+export STEPS=<number of fine-tuning steps to run> # e.g., 1000
+export PER_DEVICE_BATCH_SIZE=<batch size per device> # e.g., 1
```
Comment (A9isha) [inline docs/tutorials/grpo_with_pathways.md]: this will go unused since the training with Tunix will not honor this. and I don't think we need this for checkpoint conversion

For reference with TMVP training
  # Number of training steps.
  max_train_steps = int(
      tmvp_config.num_batches * tmvp_config.num_iterations * tmvp_config.train_fraction * tmvp_config.num_epochs
  )

File: docs/tutorials/grpo_with_pathways.md [hunk 2]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
+```bash
+python3 -m pip install torch --index-url https://download.pytorch.org/whl/cpu
+
+# using --lazy_load_tensor=true here will reduce the memory usage. eg, Llama3.1-70B conversion takes around 86GB of RAM
+python3 -m MaxText.utils.ckpt_conversion.to_maxtext src/MaxText/configs/base.yml \
+    model_name=${MODEL} \
+    hf_access_token=${HF_TOKEN} \
+    base_output_directory=${MAXTEXT_CKPT_PATH} \
+    scan_layers=True
+    --lazy_load_tensor=true
```
Comment (A9isha) [inline docs/tutorials/grpo_with_pathways.md]: for pathways, we need zarr3 and ocdbt set to false

File: docs/tutorials/grpo_with_pathways.md [hunk 3]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
 recommend creating the virtual environment outside the `maxtext` directory.
 
+
+## Setup the following environment variables before running GRPO
+
+Setup following environment variables before running GRPO
+
+```bash
+# -- Model configuration --
+export MODEL=<model name> # e.g., 'llama3.1-70b'
```
Comment (A9isha) [inline docs/tutorials/grpo_with_pathways.md]: could you please specify that `MODEL` here can be `llama3.1-8b-Instruct` version, but `MODEL` in the command below needs to be non Instruct version, for e.g, `llama3.1-8b`

This is important because for RL, most likely people would use intruction tuned models

File: docs/tutorials/grpo_with_pathways.md [hunk 4]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
+export TPU_TYPE=<tpu type> # eg. v5p-128
+export PROJECT=<gcp project name>
+export ZONE=<zone name>
+```
+
+## Get your model checkpoint
+
+You can convert a Hugging Face checkpoint to MaxText format using the `src/MaxText/utils/ckpt_conversion/to_maxtext.py` script. This is useful if you have a pre-trained model from Hugging Face that you want to use with MaxText.
+
+First, ensure you have the necessary dependencies installed. Then, run the conversion script on a CPU machine. For large models, it is recommended to use the --lazy_load_tensor flag to reduce memory usage during conversion. This command will download the Hugging Face model and convert it to the MaxText format, saving it to the specified GCS bucket.
```
Comment (A9isha) [inline docs/tutorials/grpo_with_pathways.md]: > script on a CPU machine.

is there a way to specify the CPU RAM needs in this text too? i.e., maybe rephrase the following in this text too
`# Example of converting Llama3.1-70B using --lazy_load_tensor=true which uses around 86GB of RAM`

File: docs/tutorials/grpo.md [hunk 5]
```diff
@@ -40,19 +40,64 @@ bash ~/maxtext/src/MaxText/examples/install_tunix_vllm_requirement.sh
 
 Primarily, it installs `vllm-tpu` which is [vllm](https://github.com/vllm-project/vllm) and [tpu-inference](https://github.com/vllm-project/tpu-inference) and thereby providing TPU inference for vLLM, with unified JAX and PyTorch support.
 
+## Setup the following environment variables before running GRPO
+
+Setup following environment variables before running GRPO
+
+```bash
+# -- Model configuration --
+export MODEL=<model name> # e.g., 'llama3.1-8b'
```
Comment (A9isha) [inline docs/tutorials/grpo.md]: I am thinking could you please add

export MODEL=<model name> # e.g., 'llama3.1-8b-Instruct' i.e., the name used by MaxText

File: docs/tutorials/grpo_with_pathways.md (hunk 3 above)
Comment (A9isha) [reply docs/tutorials/grpo_with_pathways.md]: I am also thinking could you please add

export MODEL=<model name> # e.g., 'llama3.1-8b-Instruct' i.e., the name used by MaxText

Comment (A9isha) [reply docs/tutorials/grpo_with_pathways.md]: I am thinking we should redefine `MODEL` and define `TOKENIZER` right before the `train_rl.py` because `TOKENIZER ` is not used in the checkpoint conversion and `MODEL` needs to called without `-Instruct` there

File: docs/tutorials/grpo_with_pathways.md (hunk 2 above)
Comment (samos123) [reply docs/tutorials/grpo_with_pathways.md]: @khatwanimohit so did it work without setting zar3 and ocdbt3 to false for you?

File: docs/tutorials/grpo_with_pathways.md (hunk 1 above)
Comment (khatwanimohit) [reply docs/tutorials/grpo_with_pathways.md]: removed

File: docs/tutorials/grpo_with_pathways.md (hunk 2 above)
Comment (khatwanimohit) [reply docs/tutorials/grpo_with_pathways.md]: done

File: docs/tutorials/grpo_with_pathways.md (hunk 3 above)
Comment (khatwanimohit) [reply docs/tutorials/grpo_with_pathways.md]: added a separate variable for HF model name and MaxText model. I think its cleaner to put all ENV variables in once place.

File: docs/tutorials/grpo_with_pathways.md (hunk 4 above)
Comment (khatwanimohit) [reply docs/tutorials/grpo_with_pathways.md]: updated

File: docs/tutorials/grpo.md (hunk 5 above)
Comment (khatwanimohit) [reply docs/tutorials/grpo.md]: done

File: docs/tutorials/grpo_with_pathways.md (hunk 3 above)
Comment (samos123) [reply docs/tutorials/grpo_with_pathways.md]: +1 this PR makes the tutorial a lot more user friendly. Can you please fill in the values for a working llama 3 70b run? So it's foolproof.

File: docs/tutorials/grpo_with_pathways.md [hunk 6]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
 recommend creating the virtual environment outside the `maxtext` directory.
 
+
+## Setup the following environment variables before running GRPO
+
+Setup following environment variables before running GRPO
+
+```bash
+# -- Model configuration --
+export HF_MODEL=<model name> # e.g., 'llama3.1-70b-Instruct'
```
Comment (samos123) [inline docs/tutorials/grpo_with_pathways.md]: Please add in the actual value that works, please do the same for all other variables.

File: docs/tutorials/grpo_with_pathways.md [hunk 7]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
+export MODEL=<model name> # e.g., 'llama3.1-70b'
+export TOKENIZER=<tokenizer path> # e.g., 'meta-llama/Llama-3.1-70B-Instruct'
+export HF_TOKEN=<Hugging Face access token>
+
+# -- MaxText configuration --
+export BASE_OUTPUT_DIRECTORY=<output directory to store run logs> # e.g., gs://my-bucket/my-output-directory
+export RUN_NAME=<name for this run> # e.g., $(date +%Y-%m-%d-%H-%M-%S)
```
Comment (samos123) [inline docs/tutorials/grpo_with_pathways.md]:
```suggestion
export RUN_NAME=llama-3-70b-run
```

File: docs/tutorials/grpo_with_pathways.md [hunk 8]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
+
+## Setup the following environment variables before running GRPO
+
+Setup following environment variables before running GRPO
+
+```bash
+# -- Model configuration --
+export HF_MODEL=<model name> # e.g., 'llama3.1-70b-Instruct'
+export MODEL=<model name> # e.g., 'llama3.1-70b'
+export TOKENIZER=<tokenizer path> # e.g., 'meta-llama/Llama-3.1-70B-Instruct'
```
Comment (samos123) [inline docs/tutorials/grpo_with_pathways.md]:
```suggestion
export TOKENIZER=;meta-llama/Llama-3.1-70B-Instruct'
```

File: docs/tutorials/grpo_with_pathways.md [hunk 9]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
+export HF_TOKEN=<Hugging Face access token>
+
+# -- MaxText configuration --
+export BASE_OUTPUT_DIRECTORY=<output directory to store run logs> # e.g., gs://my-bucket/my-output-directory
+export RUN_NAME=<name for this run> # e.g., $(date +%Y-%m-%d-%H-%M-%S)
+export MAXTEXT_CKPT_PATH=${BASE_OUTPUT_DIRECTORY}/${RUN_NAME}/0/items
+export PER_DEVICE_BATCH_SIZE=<batch size per device> # e.g., 1
+
+# -- Workload configuration --
+export WORKLOAD=<workload name>
```
Comment (samos123) [inline docs/tutorials/grpo_with_pathways.md]: Can we re-use RUN_NAME here?

File: docs/tutorials/grpo_with_pathways.md [hunk 10]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
+export TPU_CLUSTER=<cluster name>
+export TPU_TYPE=<tpu type> # eg. v5p-128
+export PROJECT=<gcp project name>
```
Comment (samos123) [inline docs/tutorials/grpo_with_pathways.md]:
```suggestion
export PROJECT_ID=<GCP Project ID>
```

Users may get confused and provide the project name which is different from project ID.

File: docs/tutorials/grpo_with_pathways.md [hunk 11]
```diff
@@ -29,6 +29,50 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
+# -- MaxText configuration --
+export BASE_OUTPUT_DIRECTORY=<output directory to store run logs> # e.g., gs://my-bucket/my-output-directory
+export RUN_NAME=<name for this run> # e.g., $(date +%Y-%m-%d-%H-%M-%S)
+export MAXTEXT_CKPT_PATH=${BASE_OUTPUT_DIRECTORY}/${RUN_NAME}/0/items
+export PER_DEVICE_BATCH_SIZE=<batch size per device> # e.g., 1
+
+# -- Workload configuration --
+export WORKLOAD=<workload name>
+export TPU_CLUSTER=<cluster name>
+export TPU_TYPE=<tpu type> # eg. v5p-128
```
Comment (samos123) [inline docs/tutorials/grpo_with_pathways.md]:
```suggestion
export TPU_TYPE=v5p-128
```

File: src/MaxText/examples/install_tunix_vllm_requirement.sh [hunk 12]
```diff
@@ -19,15 +19,11 @@
 set -e
 set -x
 
-uv pip uninstall -y jax jaxlib libtpu
+uv pip uninstall jax jaxlib libtpu
```
Comment (SurbhiJainUSC) [inline src/MaxText/examples/install_tunix_vllm_requirement.sh:22]: Did you verify it works without user input?

Comment (khatwanimohit) [reply src/MaxText/examples/install_tunix_vllm_requirement.sh]: added it back

File: docs/tutorials/grpo.md [hunk 13]
```diff
@@ -46,18 +46,63 @@ Primarily, it installs `vllm-tpu` which is [vllm](https://github.com/vllm-projec
...
+    scan_layers=True skip_jax_distributed_system=true
+
+# Example of converting Llama3.1-70B using --lazy_load_tensor=true which uses around 86GB of RAM
+
+python3 -m MaxText.utils.ckpt_conversion.to_maxtext MaxText/configs/base.yml \
+    model_name=llama3.1-70b \
+    hf_access_token=${HF_TOKEN} \
+    base_output_directory=${BASE_OUTPUT_DIRECTORY}/${RUN_NAME} \
+    scan_layers=True \
+    --lazy_load_tensors=true
```
Comment (hengtaoguo) [inline docs/tutorials/grpo.md:90]: Could you also double check if these two flags are required or not?  `hardware=cpu skip_jax_distributed_system=True`

File: docs/tutorials/grpo_with_pathways.md [hunk 14]
```diff
@@ -29,6 +29,49 @@ Furthermore, we use Pathways for [orchestration](https://cloud.google.com/ai-hyp
...
+export TPU_CLUSTER=<cluster name>
+export PROJECT_ID=<GCP project ID>
+export ZONE=<zone name>
+```
+
+## Get your model checkpoint
+
+You can convert a Hugging Face checkpoint to MaxText format using the `src/MaxText/utils/ckpt_conversion/to_maxtext.py` script. This is useful if you have a pre-trained model from Hugging Face that you want to use with MaxText.
+
+First, ensure you have the necessary dependencies installed. Then, run the conversion script on a CPU machine. For large models, it is recommended to use the --lazy_load_tensors flag to reduce memory usage during conversion, e.g., converting a Llama3.1-70B model scanned checkpoint using --lazy_load_tensors=true will use around 200GB of RAM and completes in ~10 mins. This command will download the Hugging Face model and convert it to the MaxText format, saving it to the specified GCS bucket.
```
Comment (hengtaoguo) [inline docs/tutorials/grpo_with_pathways.md]: nit: maybe break into two sentences at `e.g.`? And use quote marks around `--lazy_load_tensors`?
